pkg/collectors/dev_info_collector.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package collectors
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/callbacks"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors/devices"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/utils"
+)
+
+type DevInfoCollector struct {
+	callback      callbacks.Callback
+	devInfo       *devices.PTPDeviceInfo
+	ctx           clients.ContainerContext
+	interfaceName string
+	count         uint32
+	running       bool
+}
+
+const (
+	DevInfoCollectorName = "DevInfo"
+	DeviceInfo           = "device-info"
+	VendorIntel          = "0x8086"
+	DeviceE810           = "0x1593"
+)
+
+// Start will add the key to the running pieces of data
+// to be collects when polled
```

Comment:
While key isn't required now that we have all single purpose collectors I wanted to hold off a bit till we have a few more collectors in case we end up having to put it back in. I think that.

I'm happy to rip out keys if you think its worth doing it now.

Reply:
I removed it as it wouldn't take long to add back in.